app/src/main/java/ch/epfl/polybazaar/UI/SubmitOffer.java

```diff
@@ -0,0 +1,99 @@
+package ch.epfl.polybazaar.UI;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.google.firebase.Timestamp;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import ch.epfl.polybazaar.R;
+import ch.epfl.polybazaar.chat.ChatMessage;
+import ch.epfl.polybazaar.listing.Listing;
+import ch.epfl.polybazaar.litelisting.LiteListing;
+import ch.epfl.polybazaar.login.AuthenticatorFactory;
+
+import static ch.epfl.polybazaar.chat.ChatMessage.OFFER_ACCEPTED;
+import static ch.epfl.polybazaar.chat.ChatMessage.OFFER_PROCESSED;
+import static ch.epfl.polybazaar.listing.Listing.SOLD;
+import static java.util.UUID.randomUUID;
+
+public class SubmitOffer extends AppCompatActivity {
+
+    public static final String LISTING = "LISTING";
+    private EditText inputOffer;
+    private Listing listing;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_submit_offer);
+        inputOffer = findViewById(R.id.offer);
+        listing = (Listing)getIntent().getSerializableExtra(LISTING);
+    }
+
+    public void submitOffer(View view) {
+        if (!String.valueOf(inputOffer.getText()).isEmpty()) {
+            double offer = Double.parseDouble(inputOffer.getText().toString());

```

Comment:
What happens if the user enters non numeric input?

Reply:
The user cannot enter a non-numeric input actually